pkg/resourcemonitor/noderesourcesaggregator.go

```diff
@@ -231,14 +231,17 @@ func makeResourceMap(numaNodes int, devices []*podresourcesapi.ContainerDevices)
 	resourceMap := make(map[string]map[string]int)
 
 	for _, device := range devices {
-		deviceID2NUMAID := make(map[string]int)
+		resourceName := device.GetResourceName()
+		_, ok := resourceMap[resourceName]
+		if !ok {
+			resourceMap[resourceName] = make(map[string]int)
+		}
```

Comment:
why explicit initialization? Just curious, won't hurt

Reply:
we need to have a new map only when a device with a new resourcename is identified, in the previous case we were creating a new map for each device